plugins/woocommerce-admin/client/products/fields/attribute-field/attribute-field.tsx

```diff
@@ -15,29 +20,108 @@ import { closeSmall } from '@wordpress/icons';
 import './attribute-field.scss';
 import AttributeEmptyStateLogo from './attribute-empty-state-logo.svg';
 import { AddAttributeModal } from './add-attribute-modal';
+import { EditAttributeModal } from './edit-attribute-modal';
 import { reorderSortableProductAttributePositions } from './utils';
 
 type AttributeFieldProps = {
 	value: ProductAttribute[];
 	onChange: ( value: ProductAttribute[] ) => void;
+	productId?: number;
+};
+
+export type HydratedAttributeType = Omit< ProductAttribute, 'options' > & {
+	terms: ProductAttributeTerm[];
 };
 
 export const AttributeField: React.FC< AttributeFieldProps > = ( {
 	value,
 	onChange,
+	productId,
 } ) => {
 	const [ showAddAttributeModal, setShowAddAttributeModal ] =
 		useState( false );
+	const [ hydrationComplete, setHydrationComplete ] = useState< boolean >(
+		value ? false : true
+	);
+	const [ hydratedAttributes, setHydratedAttributes ] = useState<
```

Comment:
This will story a clone of the attributes that has the `terms` populated, as opposed to only including the `name`, which is all that is tracked on the `Product` object. We'll need term details to manage/edit them.